asserts/database.go

```diff
@@ -176,3 +223,101 @@ func (db *Database) Check(assert Assertion) error {
 	}
 	return fmt.Errorf("failed signature verification: %v", lastErr)
 }
+
+// Add persists the assertions after ensuring it is properly signed and consistent with all the stored knowledge.
+// Returns ErrAssertionNotSuperseding if trying to add a previous revision of the assertion.
+func (db *Database) Add(assert Assertion) error {
+	reg, err := checkAssertType(assert.Type())
+	if err != nil {
+		return err
+	}
+	err = db.Check(assert)
+	if err != nil {
+		return err
+	}
+	primaryKey := make([]string, len(reg.primaryKey))
+	for i, k := range reg.primaryKey {
+		keyVal := assert.Header(k)
+		if keyVal == "" {
+			return fmt.Errorf("missing primary key header: %v", k)
+		}
+		// safety against '/' etc
+		primaryKey[i] = url.QueryEscape(keyVal)
+	}
+	indexPath := filepath.Join(string(assert.Type()), filepath.Join(primaryKey...))
+	_, err = db.statEntry(assertionsRoot, indexPath)
+	switch {
+	case err == nil:
```

Comment:
The switch here for some reason seems to read a bit strangely. To help out, it feels like the logic inside this err == nil case stands on its own, so perhaps we can take it out into its own appropriately named function, and then have a shorter branches here.

What do you think?